Adds new items to the mining vendors.

TechnoAlchemisto opened the pull request:
Also, creates a tracking implanter since it apparently didn't have an implanter path before.

:cl:
rscadd: More items have been added to the mining vendor, check them out!
/:cl:

TechnoAlchemisto committed:
Adds new items to the mining vendors, also adds tracking implanter since that didn't exist before.

KazeEspada commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -120,3 +120,11 @@
+	icon_state = "stimpen"
+	volume = 70
+	amount_per_transfer_from_this = 70
+	list_reagents = list("kelotane" = 10, "bicaridine" = 10, "salbutamol" = 10, "coffee" = 20, "leporazine" = 20, "antitoxin" = 10, "epinephrine" = 10)
```
No charcoal?

KazeEspada commented on code/modules/telesci/gps.dm:
```diff
@@ -104,4 +104,9 @@ var/list/GPS_list = list()
+
+/obj/item/device/gps/mining/internal
+	icon_state = "gps-m"
+	gpstag = "MINER"
```
No way to set the tag?

TechnoAlchemisto replied:
Fraid' not. No idea how I would go around doing that for an implanter.

KazeEspada replied:
Attack self on the implanter?

TechnoAlchemisto replied:
I have no idea how to code shit, but I'll look into cooy and pasting stuff when I get back to a computer.

TechnoAlchemisto replied:
Wait a minute this makes literally no sense GPS are free

KazeEspada replied:
This is better because I can use it to powergame on station with unremovable suit sensors.

TechnoAlchemisto replied:
I'm swapping it out with a legit tracking implant

TechnoAlchemisto committed:
woops

KorPhaeron commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -120,3 +120,11 @@
 	list_reagents = list("atropine" = 10, "epinephrine" = 10, "salbutamol" = 20, "spaceacillin" = 20)
+
+/obj/item/weapon/reagent_containers/hypospray/medipen/survival
+	name = "Survival Medipen"
```
Not sure if they need this item. If they want portable quick healing they can ask chemistry

TechnoAlchemisto replied:
that same logic could justify removing the brute medkit.

Convenience should be a good incentive for mining points. It's pretty damn pricey for a one off heal. No miner is realistically going to go to chem and wait 5 minutes while they make a good good patch for them, it's a huge drop in efficiency.

TechnoAlchemisto committed:
Fixes tracking implants for miners, makes medipens not OD you on tricord.

TechnoAlchemisto committed:
Merge branch 'Survival_kit' of https://github.com/TechnoAlchemisto/-tg-station into Survival_kit

# Conflicts:
#	code/modules/reagents/reagent_containers/hypospray.dm

TechnoAlchemisto committed:
Actually doesn't OD on tricord now

octareenroon91 commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -120,3 +120,11 @@
+	icon_state = "stimpen"
+	volume = 90
+	amount_per_transfer_from_this = 90
+	list_reagents = list("kelotane" = 10, "bicaridine" = 10, "salbutamol" = 10, "coffee" = 20, "leporazine" = 20, "antitoxin" = 5, "omnizine" = 5, "epinephrine" = 10)
```
>antitoxin

That's going to flush out ALL the other meds in the syringe. Have you thought that through?

TechnoAlchemisto replied:
And here I thought that was just charcoal

TechnoAlchemisto committed:
Chemical changes.

coiax committed:
Upgraded the jaunter into the wormhole lifebelt

:cl: coiax
rscadd: The Free Golems have developed the jaunter into the  wormhole
lifebelt, which automatically teleports the user when they fall in a
chasm or suffer a major medical emergency. Must be worn on the belt to work.
/:cl:

Please don't mistake a random teleportation in crit for saving a user's
life, the Supply channel gets a notification, but suit sensors are
required to find them, and there's no guarantee the location picked is
safe (for example, the Toxins Test Site is a valid jaunt destination,
along with in the middle of the slime cages in Xenobiology).

TechnoAlchemisto committed:
Adds modification kit.

TechnoAlchemisto committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Survival_kit

TechnoAlchemisto committed:
Merge branch 'Survival_kit' of https://github.com/TechnoAlchemisto/-tg-station into Survival_kit

KorPhaeron commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -120,3 +120,11 @@
 	amount_per_transfer_from_this = 30
 	list_reagents = list("atropine" = 10, "epinephrine" = 10, "salbutamol" = 20, "spaceacillin" = 20)
+
+/obj/item/weapon/reagent_containers/hypospray/medipen/survival
```
Our conversation vanished but

1) I agree this is obviously more efficient than asking chemistry for stuff

2) I'm worried though because "I can't rely on anyone else, it takes too much time" is what led to every job being a self sufficient R-walled box where everyone is bored to death

KorPhaeron commented on code/__DEFINES/combat.dm:
```diff
@@ -70,3 +70,13 @@
+ #define SAWN_SAWING -1
+
+ #define WEAPON_LIGHT 0
+ #define WEAPON_MEDIUM 1
```
?

coiax committed:
Golem ship now has its own /area

And a bluespace beacon.

coiax committed:
5 tick medijaunt, no radio, golem jaunters + areas

The number of ticks before a medical emergency jaunt happens is now
five. No radio message is produced.

Free Golems now have no restrictions on what they can purchase. Wormhole
jaunters now specifically drop golems near their ship.

coiax commented:
Could you make the KA upgrader give the KA a wide trigger? That would be `trigger_guard = TRIGGER_GUARD_ALLOW_ALL`. Would allow Free Golems to modify KAs so they can use them.

TechnoAlchemisto commented:
I don't see why not

TechnoAlchemisto committed:
Modkits remove the race restrictions from KAs.

coiax committed:
Buffs buffs buffs

I'd say something, but I'm sure I'd regret it later.

Incomptinence commented:
Looks good and while it has a steep price tag I assume the mod kit will let golems use accelerators? 

Neat if so.

coiax commented:
Golems can't buy them at the moment, but I'm working on that.

TechnoAlchemisto commented:
You should check my newest commit again @coiax

coiax commented:
tbh, we should switch the free golem mining vendor to have all the standard equipment + some bonus goods. I've got some code for jaunters that'll mean that free golems that use them will get sent to the ship.

coiax committed:
Merge branch 'wormhole-lifebelt' into coiax-jaunter-changes

Conflicts:
	code/modules/mining/equipment_locker.dm

KorPhaeron commented on code/modules/mining/equipment_locker.dm:
```diff
@@ -1134,12 +1138,15 @@
 		new /datum/data/mining_equipment("Sulphuric Acid",			/obj/item/weapon/reagent_containers/glass/beaker/sulphuric,        		500),
 		new /datum/data/mining_equipment("Brute First-Aid Kit",		/obj/item/weapon/storage/firstaid/brute,						   		600),
 		new /datum/data/mining_equipment("Advanced Scanner",		/obj/item/device/t_scanner/adv_mining_scanner,                     		800),
+		new /datum/data/mining_equipment("Kinetic Accelerator", 	/obj/item/weapon/gun/energy/kinetic_accelerator,               	   		750),
```
Costs 1/3 as much as the super KA but doubles your damage.

Also even less reason for science to make you anything.

TechnoAlchemisto commented:
Jesus christ adding anything desirable to the mining vendor is hell, I'm having trouble thinking up ways to reward miners with cool items without stepping on other department interactions. Maybe the concept of mining point redemption being good is flawed, exploring is so much more rewarding than spending your points.

You have a point, I'm going to make some modifications to this PR, namely removing the mod kit from human miners, and leaving it for golems but up their KA cost that would mane an incentive for some sort of interaction between miners and free golems.

coiax commented:
If something is OP, make the price higher until it's no longer OP.

coiax committed:
Advanced lifebelts, golem superiority

TechnoAlchemisto committed:
Merge pull request #1 from coiax/coiax-jaunter-changes

Jaunter changes, golem ship has area and tracking beacon, equipment

TechnoAlchemisto committed:
Removes modkit from minervendors, adds modkit to RnD!

KorPhaeron commented on code/modules/mining/equipment_locker.dm:
```diff
@@ -533,41 +620,70 @@
+			return TRUE
+	return FALSE
+
+/obj/item/device/wormhole_lifebelt/advanced
```
zombie pr aaaaaaa

coiax replied:
NO ESCAPE. But seriously, it costs like 2000 now, we can still disable it if you hate it that much.

My PR also made free golems able to buy jaunters but not use them to escape to the station, which is good.

TechnoAlchemisto committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Survival_kit

# Conflicts:
#	code/game/objects/items.dm

TechnoAlchemisto commented:
fuck

TechnoAlchemisto committed:
COMPILE DAMN YOU!!

TechnoAlchemisto committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Survival_kit

# Conflicts:
#	code/game/objects/items.dm
#	code/modules/mining/equipment_locker.dm

TechnoAlchemisto commented:
ha...
hahaha........

coiax commented:
Yes.

coiax committed:
I made it compile

TechnoAlchemisto committed:
Merge pull request #2 from coiax/Survival_kit

I made it compile

TechnoAlchemisto commented:
Coiax fixed it, we gucci.

KorPhaeron commented on code/modules/mining/equipment_locker.dm:
```diff
@@ -500,38 +517,108 @@
-	if(!L.len)
+
+	// If no beacons found for golems, home in on dat gold statue.
+	if(!L.len && golem_mode)
```
This whole bit strikes me as snowflake and wrong also I still think the item is HUG BOX but I'll sleep on it

coiax replied:
Look at the stats Kor, no one is using jaunters for chasm protection at the moment. We can remove the gold statue check if you're unhappy with it.

TechnoAlchemisto commented:
Kor "Sleep On it" Phaeron strikes again. This PR is no longer Merge Conflict or WIP if someone would please remove the tags :^)

TechnoAlchemisto committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Survival_kit

# Conflicts:
#	code/game/objects/items.dm

TechnoAlchemisto committed:
Honor to my country, not to you :(

KorPhaeron commented on code/game/objects/items/devices/modificationkit.dm:
```diff
@@ -0,0 +1,23 @@
+/obj/item/modkit
```
this could probably have a better item path, its a very general path for a very specific item

KorPhaeron commented on code/game/objects/items.dm:
```diff
@@ -575,7 +575,13 @@ obj/item/proc/item_action_slot_check(slot, mob/user)
+
 /obj/item/proc/can_dismember()
+
+	return sharpness && w_class >= 3
```
uhm you brke this

TechnoAlchemisto committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Survival_kit

TechnoAlchemisto committed:
fuck

TechnoAlchemisto committed:
h-haha?

KorPhaeron commented on code/modules/mining/equipment_locker.dm:
```diff
@@ -426,7 +429,7 @@
 
 /obj/machinery/mineral/equipment_vendor/proc/RedeemVoucher(obj/item/weapon/mining_voucher/voucher, mob/redeemer)
-	var/selection = input(redeemer, "Pick your equipment", "Mining Voucher Redemption") as null|anything in list("Survival Capsule and Explorer Belt", "Resonator", "Mining Drone", "Advanced Scanner")
+	var/selection = input(redeemer, "Pick your equipment", "Mining Voucher Redemption") as null|anything in list("Survival Capsule and Explorer Belt", "Resonator", "Mining Drone", "Advanced Scanner", "Wormhole Lifebelt")
```
You forgot to remove it from the list here

KorPhaeron commented on code/modules/mining/equipment_locker.dm:
```diff
@@ -487,6 +490,8 @@
 
 /**********************Jaunter**********************/
 
+/**********************Jaunter**********************/
```
doubled header

TechnoAlchemisto committed:
Kor you know I would never actually ban you

TechnoAlchemisto committed:
Merge branch 'Survival_kit' of https://github.com/TechnoAlchemisto/-tg-station into Survival_kit

KorPhaeron commented on code/game/objects/items/devices/modificationkit.dm:
```diff
@@ -0,0 +1,23 @@
+	origin_tech = "programming=2;materials=2;magnets=4"
+	var/uses = 1
+
+/obj/item/modkit/afterattack(var/obj/item/weapon/gun/energy/kinetic_accelerator/C, mob/user)
```
no var/ in proc arguments

just

/obj/item/weapon/gun/energy/kinetic_accelerator/C

I wish I could just make these changes and merge it myself I think there is a way to do it

TechnoAlchemisto replied:
I mean technically you could PR to this branch like coiax did, but this works as it is right now and I recall it not working at all until oranges helped me and told me to put the var there.

TechnoAlchemisto committed:
i could personally care less for HBls adminning style

TechnoAlchemisto commented:
Your ban has been lifted kor thanks for the business

coiax commented:
You added jaunters back to the Golem Ship without including any of my jaunters-target-the-golem-ship-for-golems stuff.

coiax commented:
I mean, jeez, you only had to take the medical lifebelt out of the vendor, you didn't have to revert the whole damn thing.

TechnoAlchemisto commented:
I'm just following orders
